### PR TITLE
Default scheduler handling for storIOContentResolver

### PR DESCRIPTION
**Pull request by nikitin-da**
The second part of #511

### CODE REVIEW

File: storio-content-resolver/src/main/java/com/pushtorefresh/storio/contentresolver/impl/DefaultStorIOContentResolver.java [hunk 1]
```diff
@@ -139,6 +155,9 @@ public CompleteBuilder contentResolver(@NonNull ContentResolver contentResolver)
         @Nullable
         private Handler contentObserverHandler;
 
+        @Nullable
+        private Scheduler defaultScheduler = RX_JAVA_IS_IN_THE_CLASS_PATH ? Schedulers.io() : null;
```
Comment (artem-zinnatullin) [inline storio-content-resolver/src/main/java/com/pushtorefresh/storio/contentresolver/impl/DefaultStorIOContentResolver.java]: 👍

File: storio-content-resolver/src/main/java/com/pushtorefresh/storio/contentresolver/operations/delete/PreparedDeleteByQuery.java [hunk 2]
```diff
@@ -102,16 +102,16 @@ public DeleteResult executeAsBlocking() {
     public Observable<DeleteResult> asRxObservable() {
         throwExceptionIfRxJavaIsNotAvailable("asRxObservable()");
 
-        return Observable
-                .create(OnSubscribeExecuteAsBlocking.newInstance(this))
-                .subscribeOn(Schedulers.io());
+        final Observable<DeleteResult> observable =
+                Observable.create(OnSubscribeExecuteAsBlocking.newInstance(this));
+        return RxJavaUtils.subscribeOn(observable, storIOContentResolver);
```
Comment (artem-zinnatullin) [inline storio-content-resolver/src/main/java/com/pushtorefresh/storio/contentresolver/operations/delete/PreparedDeleteByQuery.java]: Can't you create a function that will create `Observable` from current operation and apply scheduler? Will save 2 lines in each operation.

Same is applicable for `Completable` and `Single`, I believe :)

Comment (nikitin-da) [inline storio-content-resolver/src/main/java/com/pushtorefresh/storio/contentresolver/operations/delete/PreparedDeleteByQuery.java]: Okay =)

File: storio-content-resolver/src/test/java/com/pushtorefresh/storio/contentresolver/impl/DefaultStorIOContentResolverTest.java [hunk 3]
```diff
@@ -257,4 +260,34 @@ public TypeMappingFinder typeMappingFinder() throws NoSuchFieldException, Illega
...
         }
     }
+
+    @Test
+    public void defaultSchedulerReturnsIOSchedulerIfNotSpecified() {
+        StorIOContentResolver storIOContentResolver = DefaultStorIOContentResolver.builder()
+                .contentResolver(mock(ContentResolver.class))
+                .build();
+
+        assertThat(storIOContentResolver.defaultScheduler()).isEqualTo(io());
```
Comment (artem-zinnatullin) [inline storio-content-resolver/src/test/java/com/pushtorefresh/storio/contentresolver/impl/DefaultStorIOContentResolverTest.java]: `isSameAs()`

File: storio-content-resolver/src/test/java/com/pushtorefresh/storio/contentresolver/impl/DefaultStorIOContentResolverTest.java [hunk 4]
```diff
@@ -257,4 +260,34 @@ public TypeMappingFinder typeMappingFinder() throws NoSuchFieldException, Illega
...
+
+    @Test
+    public void defaultSchedulerReturnsSpecifiedScheduler() {
+        Scheduler scheduler = mock(Scheduler.class);
+        StorIOContentResolver storIOContentResolver = DefaultStorIOContentResolver.builder()
+                .contentResolver(mock(ContentResolver.class))
+                .defaultScheduler(scheduler)
+                .build();
+
+        assertThat(storIOContentResolver.defaultScheduler()).isEqualTo(scheduler);
```
Comment (artem-zinnatullin) [inline storio-content-resolver/src/test/java/com/pushtorefresh/storio/contentresolver/impl/DefaultStorIOContentResolverTest.java]: `isSameAs()`

File: storio-content-resolver/src/test/java/com/pushtorefresh/storio/contentresolver/impl/DefaultStorIOContentResolverTest.java [hunk 5]
```diff
@@ -257,4 +260,34 @@ public TypeMappingFinder typeMappingFinder() throws NoSuchFieldException, Illega
...
+    }
+
+    @Test
+    public void defaultSchedulerReturnsNullIfSpecifiedSchedulerNull() {
```
Comment (artem-zinnatullin) [inline storio-content-resolver/src/test/java/com/pushtorefresh/storio/contentresolver/impl/DefaultStorIOContentResolverTest.java]: 👍

File: storio-content-resolver/src/test/java/com/pushtorefresh/storio/contentresolver/operations/SchedulerChecker.java [hunk 6]
```diff
@@ -0,0 +1,62 @@
...
+    @NonNull
+    private final TestScheduler scheduler;
+
+    private SchedulerChecker(@NonNull StorIOContentResolver storIOContentResolver) {
+        this.storIOContentResolver = storIOContentResolver;
+        scheduler = test();
+        Mockito.when(storIOContentResolver.defaultScheduler()).thenReturn(scheduler);
+    }
+
+    public static SchedulerChecker create(@NonNull StorIOContentResolver storIOContentResolver) {
```
Comment (artem-zinnatullin) [inline storio-content-resolver/src/test/java/com/pushtorefresh/storio/contentresolver/operations/SchedulerChecker.java]: nit: `@NonNull` for return type

File: storio-content-resolver/src/test/java/com/pushtorefresh/storio/contentresolver/operations/SchedulerChecker.java [hunk 7]
```diff
@@ -0,0 +1,62 @@
...
+        operation.asRxSingle().subscribe();
+        check(operation);
+    }
+
+    public void checkAsCompletable(@NonNull PreparedWriteOperation operation) {
+        operation.asRxCompletable().subscribe();
+        check(operation);
+    }
+
+    private void check(@NonNull PreparedOperation operation) {
```
Comment (artem-zinnatullin) [inline storio-content-resolver/src/test/java/com/pushtorefresh/storio/contentresolver/operations/SchedulerChecker.java]: 👍

**Comment by artem-zinnatullin** [comment]
👍 (few nits)

Dude this is crazy! 
screen shot 2016-07-20 at 23 50 22

**Comment by nikitin-da** [comment]
@artem-zinnatullin hehe, but 0.1% decrease :crying_cat_face:

**Pull request merged**